File: 0797-all-paths-from-source-to-target/0797-all-paths-from-source-to-target.cpp
class Solution {
public:
    
    vector<vector<int>>ans;
    void dfs(vector<vector<int>>graph,int idx,vector<int>path){
        
            //cout<<idx<<endl;
            path.push_back(idx);
            if(idx==graph.size()-1){
                    //cout<<idx<<endl;
                    ans.push_back(path);
                    return;
            }        
            
            for(int i=0;i<graph[idx].size();i++){
                 dfs(graph,graph[idx][i],path);            
            }
        
            
    }
    
    
    
    
    vector<vector<int>> allPathsSourceTarget(vector<vector<int>>& graph) {
        
        
            vector<int>path;
            dfs(graph,0,path);
            return ans;
        
        
    }
};

File: 0797-all-paths-from-source-to-target/0797-all-paths-from-source-to-target.py
class Solution:
    
    def __init__(self):
        
            self.ans=[]
    
    
    
    def helper(self,graph,indx,dest,visited,trav):
        
                
                if visited[indx]==True:
                    return 
        
                if indx==dest:
                        
                        trav.append(indx)
                        self.ans.append(trav[:])
                        trav.pop(-1)
                        return
                
                
                visited[indx]=True
                trav.append(indx)
                for item in graph[indx]:
                    self.helper(graph,item,dest,visited,trav)
                    
        
                visited[indx]=False
                trav.pop(-1)
               
                
                
        
        
        
        
        
        
    
    
        
    
    
    
    
    def allPathsSourceTarget(self, graph: List[List[int]]) -> List[List[int]]:
        
        n=len(graph)
        visited=[False for i in range(len(graph))]
        self.helper(graph,0,n-1,visited,[])
        return self.ans